src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -0,0 +1,65 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Dapr.Client;
+using Microsoft.Extensions.Configuration;
+
+namespace Dapr.Extensions.Configuration.DaprSecretStore
+{
+    /// <summary>
+    /// A Dapr Secret Store based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal class DaprSecretStoreConfigurationProvider : ConfigurationProvider
+    {
+        private readonly string store;
+
+        private readonly IEnumerable<DaprSecretDescriptor> secretDescriptors;
+
+        private readonly DaprClient client;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="secretDescriptors">The secrets to retrieve.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(string store, IEnumerable<DaprSecretDescriptor> secretDescriptors, DaprClient client)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(store, nameof(store));
+            ArgumentVerifier.ThrowIfNull(secretDescriptors, nameof(secretDescriptors));
+            ArgumentVerifier.ThrowIfNull(client, nameof(client));
+
+            this.store = store;
+            this.secretDescriptors = secretDescriptors;
+            this.client = client;
+        }
+
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync()
+        {
+            var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var secretDescriptor in secretDescriptors)
+            {
+                var result = await client.GetSecretAsync(store, secretDescriptor.SecretName, secretDescriptor.Metadata).ConfigureAwait(false);
+
+                foreach (var key in result.Keys)
+                {
+                    if (data.ContainsKey(key))
+                    {
+                        throw new FormatException($"A duplicate key '{key}' was found.");
```

Comment:
This should not be a format exception, consider throwing a more meaningful one. Also please make text more descriptive of the problem.

Reply:
@amanbha this is the behavior of the `JsonConfigurationProvider` (https://github.com/dotnet/extensions/blob/master/src/Configuration/Config.Json/src/JsonConfigurationFileParser.cs#L81) you mentioned in a previous conversation. But I see your point here.

I could just change the message (keep consistent with the other provider):

``` chsharp
throw new FormatException($"A duplicate key '{key}' was found in the secret store {store}.");
```

or I could throw an `ArgumentException` with the new message:

``` chsharp
throw new ArgumentException($"A duplicate key '{key}' was found in the secret store {store}.");
```
`ArgumentException` is commonly used to indicate a developer error (configuration error in this case)

Thoughts?

Reply:
@cmendible Sure, will InvalidOperationException be the right thing to do here? https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=netcore-3.1#remarks

Reply:
InvalidOperationException seems to a better fit here. Fixed in 5ef3ef1

Reply:
@cmendible Can this exception message also tell what a user can do to fix this?

Reply:
@amanbha new messages statest that there is a duplicate key. How should we address the dev here? 

$"A duplicate key '{key}' was found in the secret store '{store}'. Please remove any duplicates."

Or should we leave it as is?

Reply:
@cmendible  This would be good: $"A duplicate key '{key}' was found in the secret store '{store}'. Please remove any duplicates from your secret store."